Makefile

```diff
@@ -34,10 +34,17 @@ TEST_DIRS     ?= $(shell sh -c "find $(TOP_SRC_DIRS) -name \\*_test.go \
                    -exec dirname {} \\; | sort | uniq")
 VERSION       ?= $(shell git describe --always --abbrev=7 --dirty)
 ifeq ($(shell uname -s),Darwin)
+# on osx run stat against /dev/null and check if it has any stdout
+# output.  if stdout is blank, bsd-stat because it returned an error
+# to stderr. Otherwise use gnu-stat args.
+ifeq ($(shell stat -c"%U" /dev/null 2> /dev/null),)
 STAT           = stat -f '%c %N'
 else
 STAT           = stat -c '%Y %n'
 endif
+else
+STAT           = stat -c '%Y %n'
+endif
```

Comment:
I would recommend simplifying all this. The bit about being Darwin or not isn't relevant in comparison to simply knowing whether we're dealing with gnu or bsd stat. i.e. `ifeq ($(shell stat -c"%U" /dev/null 2> /dev/null),)`, should always happen be evaluated regardless of OS.